### PR TITLE
[Messenger] Add `--all` option to `messenger:consume`

### LINKED ISSUE #52364: [Messenger] Provide a way to consume all transports without specifying/knowing their names
### Description

When using https://github.com/php-enqueue/enqueue, a command:
```
enqueue:consume
```
is provided, in which there is no need to specify the consumers.

But when I run 
```
bin/console messenger:consume
```
I'm getting prompt about which transport I want to consume.

I saw in the code there was
```
$defaultReceiverName = 1 === \count($this->receiverNames) ? current($this->receiverNames) : null;
$defaultReceivers = $defaultReceiverName ? [$defaultReceiverName] : [];
```

What about something like --all in order to consume everything ? Would it be a good idea ?
Or maybe a commande should be added to list all the transports and then the output could be used for the consume command (or even to setup different supervisor with a foreach loop on the output ?)

I'd like to avoid updating my conf/commands
```
bin/console messenger:consume foo1,foo2,...
```
everytime.

### Example

_Originally posted by @VincentLanglet in https://github.com/symfony/symfony/discussions/51989_

### PR DESCRIPTION
**Pull request by ghost**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| Issues        | Closes #52364 
| License       | MIT

When implementing this feature the problem with sync transports came out so the `if` statement for this was needed. I can see someone reported this 2 months ago in https://github.com/symfony/symfony/issues/51556. I think this be can fixed properly in a dedicated PR because it requires to dig into MessengerPass I guess.

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]
Looks good to me

File: src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php [hunk 1]
```diff
@@ -135,6 +141,10 @@ protected function interact(InputInterface $input, OutputInterface $output)
     {
         $io = new SymfonyStyle($input, $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output);
 
+        if ($input->getOption('all')) {
+            return;
+        }
+
         if ($this->receiverNames && !$input->getArgument('receivers')) {
             $io->block('Which transports/receivers do you want to consume?', null, 'fg=white;bg=blue', ' ', true);
```
Comment (OskarStark) [inline src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php:147]: we can add a hint here (line 153) to also use the `all` option

**Comment by OskarStark** [comment]
This must target 7.1 due to feature freeze period

File: src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php (hunk 1 above)
Comment (ghost) [reply src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php:147]: Hi. I started adding this but then I realized that the hint in line 153 relates to how to interact with the question (see the `e.g. receiver1, receiver2` part which works only when answering question, not when we pass multiple receivers as command arguments [commas are not allowed here]).

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @javaDeveloperKid.

**Comment by Wirone** [comment]
Can you consider this as a bug fix and backport it to 6.4 😂 (jk)? This exactly something we need in our app, where we migrate maaaany workers using multiple transports (because our AMQP instance uses multiple exchanges). It would be much better to use `--all` instead of specifying all the transports (receivers) in the CLI command.

**Comment by alexisLefebvre** [comment]
> Can you consider this as a bug fix and backport it to 6.4 😂 (jk)? This exactly something we need in our app, where we migrate maaaany workers using multiple transports (because our AMQP instance uses multiple exchanges). It would be much better to use `--all` instead of specifying all the transports (receivers) in the CLI command.

What if you use 7.1 in your app?

```shell
composer require symfony/messenger:^7.1
```

Update: I just realized that there is a `7.1` branch but it isn't released yet. It will be [released in May](https://symfony.com/releases/7.1), then upgrading should work and allow you to use the `--all` option.

In the meantime, using a commit or the Composer flag `@dev` for this package may work.

**Comment by Wirone** [comment]
@alexislefebvre Thanks for the feedback, but actually I knew it before ☺️. I've added `@TODO` for migrating to 7.1 when it's available and then we'll use `--all`. Unfortunately `@dev` is not suitable for our system, which is too big and too important.

**Comment by ghost** [comment]
@Wirone Another approach would be to copy past this class from 7.1 branch to your respository and register this in your composer,json. Composer will pick up the first class found in this namespace which will be yours. 
When you upgrade to 7.1+ one day, you will just remove this class.

P.S. Remember to check that changes between your 6.4.x locked in your composer.lock and 7.1 branch are only the ones from this PR. If not then decide if it's safe to use it.

**Comment by Wirone** [comment]
FYI: I was able to introduce `--all` in Symfony 6.4 pretty easily - [here's how](https://blog.codito.dev/2024/04/messenger-consume-all-transports-on-symfony-6.4/) 😎.

**Pull request merged**